classwork/controlWork1/priorityQueue/priorityQueue/PriorityQueueParallel.cs

```diff
@@ -0,0 +1,71 @@
+﻿public class PriorityQueueParallel<TPriority, TValue> where TPriority : IComparable
+{
+    private readonly object _lockObject = new();
+    private readonly SortedDictionary<TPriority, Queue<TValue>> _buffer;
+    private int _size;
+
+    /// <summary>
+    /// Getting size of the queue.
+    /// </summary>
+    public int Size
+    {
+        get
+        {
+            lock (_lockObject)
+            {
+                return _size;
+            }
+        }
+    }
```

Comment:
```suggestion
    public int Size => _size;
```
Здесь `lock` не нужен. `int` атомарен (а значит другие потоки нам не помешают прочитать корректное значение), и мы не меняем значение переменной (а значит наш поток не помешает другим).

Reply:
-0.5